src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -1282,9 +1294,9 @@ static void addBsonFilter
   }
   else
   {
-    std::string details = std::string("unknown query operator: ") + opr;
+    std::string details = std::string("unknown query operator: '") + opr + "'";
     alarmMgr.badInput(clientIp, details);
-    pushBackFilter = false;
+    return false;
   }
 
   if (pushBackFilter)
```

Comment:
Given the early-return which formarly corresponds to `pushBackFilter = false`, this if could be removed, replaced by just: 

```
filters.push_back(f);
```

As side effect, the `pushBackFilter` variable is no longer needed and can be completely removed from this function.